### PR TITLE
Also include additional tables in additional rows computation

### LINKED ISSUE #86: Number of additional rows is not determined correctly
We seem to be missing a case when a file is generated that is not present in the "ground truth". I.e. we do not count those rows as additional rows. `PRC_VINT` in e.g. Demo 4 is a good example...

### PR DESCRIPTION
**Pull request by siddharth-krishna**
Fixes #86

### CODE REVIEW

File: times_reader/main.py [hunk 1]
```diff
@@ -199,28 +199,37 @@ def compare(
...
+
     total_gt_rows = 0
     total_correct_rows = 0
-    total_additional_rows = 0
     for table_name, gt_table in sorted(
-        ground_truth.items(), reverse=True, key=lambda t: len(t[1].values)
+        ground_truth.items(), reverse=True, key=lambda t: len(t[1])
     ):
-        total_gt_rows += len(gt_table.values)
+        total_gt_rows += len(gt_table)
```
Comment (siddharth-krishna) [inline times_reader/main.py:216]: `.values` unnecessarily converts the DataFrame to a numpy `ndarray`.

File: times_reader/main.py [hunk 2]
```diff
@@ -199,28 +199,37 @@ def compare(
...
                     f"WARNING: Table {table_name} header incorrect, was"
-                    f" {data_table.columns.values}, should be {transformed_gt_cols}"
+                    f" {data_cols}, should be {transformed_gt_cols}"
                 )
             else:
                 # both are in string form so can be compared without any issues
-                gt_rows = set(tuple(row) for row in gt_table.values.tolist())
-                data_rows = set(tuple(row) for row in data_table.values.tolist())
+                gt_rows = set(tuple(row) for row in gt_table.to_numpy().tolist())
+                data_rows = set(tuple(row) for row in data_table.to_numpy().tolist())
```
Comment (siddharth-krishna) [inline times_reader/main.py:232]: And `.values` has been deprecated, so using `.to_numpy` instead

File: utils/run_benchmarks.py [hunk 3]
```diff
@@ -232,7 +232,7 @@ def run_all_benchmarks(
     if args.run is not None:
         benchmark = next((b for b in spec["benchmarks"] if b["name"] == args.run), None)
         if benchmark is None:
-            print(f"ERROR: could not find {args.run} in {args.benchmark_yaml}")
+            print(f"ERROR: could not find {args.run} in {args.benchmarks_yaml}")
```
Comment (siddharth-krishna) [inline utils/run_benchmarks.py:235]: Found this typo while testing this PR

**Review by siddharth-krishna — COMMENTED** [review]
If this change looks okay, we will need to force merge this because it increases additional rows, so will fail CI.

File: times_reader/main.py [hunk 4]
```diff
@@ -199,28 +199,37 @@ def compare(
...
         print(f"WARNING: Missing {len(missing)} tables: {missing_str}")
 
+    additional_tables = set(data.keys()) - set(ground_truth.keys())
+    additional_str = ", ".join(
+        [f"{x} ({data[x].shape[0]})" for x in sorted(additional_tables)]
+    )
+    if len(additional_tables) > 0:
+        print(f"WARNING: {len(additional_tables)} additional tables: {additional_str}")
+    # Additional rows starts as the sum of lengths of additional tables produced
+    total_additional_rows = sum(len(data[x]) for x in additional_tables)
```
Comment (olejandro) [inline times_reader/main.py:209]: @siddharth-krishna, what will this result in if `additional_tables` is empty?

**Review by olejandro — APPROVED** [review]

File: times_reader/main.py (hunk 4 above)
Comment (siddharth-krishna) [reply times_reader/main.py:209]: `sum([]) == 0` :)

**Comment by siddharth-krishna** [comment]
Getting closer to passing CI! :)

I couldn't figure out how to not generate attribute tables that don't appear in "MainData", without adding more special case code for attributes, so I've just dropped any table that's empty after filtering for now. Happy to revisit this later if this causes issues.

Here are the remaining additional tables that are generated. Let me know if these can also be fixed via a filter:
```
% find benchmarks/out -name stdout | xargs grep 'additional tables' | sort
benchmarks/out/DemoS_001-all/stdout:WARNING: 1 additional tables: PASTYEAR (1)
benchmarks/out/DemoS_002-all/stdout:WARNING: 1 additional tables: PASTYEAR (1)
benchmarks/out/DemoS_003-all/stdout:WARNING: 1 additional tables: PASTYEAR (1)
benchmarks/out/DemoS_004-all/stdout:WARNING: 1 additional tables: PASTYEAR (1)
benchmarks/out/DemoS_004/stdout:WARNING: 1 additional tables: PASTYEAR (1)
benchmarks/out/DemoS_004a/stdout:WARNING: 1 additional tables: PASTYEAR (1)
benchmarks/out/DemoS_004b/stdout:WARNING: 1 additional tables: PASTYEAR (1)
benchmarks/out/DemoS_005-all/stdout:WARNING: 1 additional tables: PASTYEAR (1)
benchmarks/out/DemoS_006-all/stdout:WARNING: 1 additional tables: PASTYEAR (1)
benchmarks/out/DemoS_007-all/stdout:WARNING: 1 additional tables: PASTYEAR (1)
benchmarks/out/DemoS_008-all/stdout:WARNING: 1 additional tables: PASTYEAR (1)
benchmarks/out/Ireland/stdout:WARNING: 3 additional tables: COM_BNDNET (1), COM_PKTS (3), NCAP_PKCNT (2)
```

**Pull request merged**